src/components/text-editor/prosemirror-adapter/plugins/table-plugin.ts

```diff
@@ -0,0 +1,35 @@
+import { tableNodes, tableEditing } from 'prosemirror-tables';
+
+export const tableEditingPlugin = tableEditing;
+
+export const getTableNodes = () => {
+    return tableNodes({
+        tableGroup: 'block',
+        cellContent: 'block+',
+        cellAttributes: {
+            background: {
+                default: null,
+                getFromDOM: (dom) => {
+                    return dom.style.backgroundColor || null;
+                },
+                setDOMAttr: (value: string, attrs: Record<string, string>) => {
+                    if (value) {
+                        attrs.style =
+                            (attrs.style || '') + `background-color: ${value};`;
```

Comment:
We probably need to check that we aren't opening up a vulnerability to script injection here. We might need to sanitise the value before using it.

Reply:
I did a quick test, and it seems like any tags are encoded if not explicitly allowed
![image](https://github.com/user-attachments/assets/2c2a2367-ff6c-4055-88b3-d3302bfaf559)

Also tried to research a bit, and it seems that the prosemirror developers are generally quite on top of protecting against these kind of things, such as discussed in this forum post https://discuss.prosemirror.net/t/heads-up-xss-risk-in-domserializer/6572.

I do feel a bit out of my comfort zone with this topic though, so I would appreciate a second opinion :)

Reply:
Yeah, it sounds very reasonable that prosemirror would come with good protections built-in, or at least possible to easily enable (in which case I hope we've done so). What I'm concerned about is if directly manipulating DOM attributes like this, in a plugin, bypasses the built-in protections? I'm a bit unsure where the value of the `value` parameter comes from? It from the pasted content originally, of course, but has that all first gone through sanitising first, or are we reading it straight from the raw input?

You don't have to be the one who can answer these questions @DavidAkkerman. I'm guessing that @john-traas may have already looked into stuff like this in his previous work on the text editor, and if not, I, John, or someone else from our team can look it up 🙂